File: gestion-proyectos/src/main/java/com/arcticcuyes/gestion_proyectos/dto/Proyecto/TareaDTO.java
package com.arcticcuyes.gestion_proyectos.dto.Proyecto;

import java.sql.Timestamp;
import java.util.List;
import java.util.Set;

import jakarta.validation.constraints.NotNull;
import lombok.*;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class TareaDTO {
    private Long id;
    private String titulo;
    private String descripcion;
    private Timestamp fechaInicio;
    private Timestamp fechaFin;
    private List<SubtareaDTO> subtareas;
    private Long estado;
    private List<Long> participantesAsignados;
}

File: client/src/types/proyecto/Tarea/dto/TareaDTO.ts
export type TareaDTO = {
  idTarea?: string;
  titulo: string;
  descripcion?: string;
  fechaInicio: Date;
  fechaFin: Date;
  createdAt?: string;
  updatedAt?: string;
  tareaAnterior?: TareaDTO;
  estado: number;
  participantesAsignados?: string[]; // Consultores
  subtareas?: SubTareaDTO[];
};

export type SubTareaDTO = {
  descripcion: string;
  completado: boolean;
};
